### PR TITLE
fix(developer): handle paste of TSV into Wordlist grid

### PR DESCRIPTION
**Pull request by mcdurdin**
When pasting into the wordlist grid, the editor will now check if the clipboard text contains newlines or tabs, and if so, will paste the text as new rows in the wordlist (with a prompt), rather than pasting the whole text into the current cell.

If the clipboard text does not contain NL or TAB then the default in-cell editor paste handling will run.

Fixes: #12517

# User Testing

These tests should be run in the Wordlist grid view in the Lexical Model Editor. Create a new lexical model and open it in the Lexical Model Editor

* **TEST_PASTE_SIMPLE_STRING:** Copy a single word into the clipboard, and then paste it into a cell in the wordlist editor. It should just paste the word in without errors.

* **TEST_PASTE_MULTILINE_STRING:** Copy the text below to the clipboard, and then paste it into a cell in the wordlist editor. You should be prompted to add new rows to the wordlist, and the text below should appear in the grid as new rows when you click Yes.

```
TŦE	13644
E	9134
SEN	4816
Ȼ	3479
SW̱	2621
NIȽ	2314
U¸	2298
I¸	1988
ȻSE	1925
I	1884
```

### CODE REVIEW

File: developer/src/tike/child/Keyman.Developer.UI.UframeWordlistEditor.pas [hunk 1]
```diff
@@ -217,6 +223,48 @@ procedure TframeWordlistEditor.EnableControls;
...
+  Inc(FSetup);
+  try
+    try
+      Value := Clipboard.AsText;
+    except
+      on E:EClipboardException do
+        Exit(False);
+    end;
+
+    if not Value.Contains(#9) and not Value.Contains(#13) then
```
Comment (darcywong00) [inline developer/src/tike/child/Keyman.Developer.UI.UframeWordlistEditor.pas:252]: I'm assuming magic numbers for tab and carriage return, right?
Should we also handle line feed?

Comment (mcdurdin) [reply developer/src/tike/child/Keyman.Developer.UI.UframeWordlistEditor.pas:252]: I guess we could handle #10 as well but it's not really ever used without #13 in Windows.

**Comment by dinakaranr** [comment]
# Test Results
I tested this issue with the attached "18.0.133-alpha-test-12594" build on the Windows 10 OS. Here is my observation.

* **TEST_PASTE_SIMPLE_STRING (Passed):** 
1. I installed the keymandeveloper-18.0.133.exe build in the Windows 10.
2. Created a new "Lexical Model" project.
3. Navigate to the "Models" tab and then click the "*.model.ts" link button.
4. Copy a word from a notepad file.
5. Click in the "Add word..." box.
6. Paste the text by pressing the CTRL+V key.
7. Verified that the word appears and no errors are displayed.
It works well.

* **TEST_PASTE_MULTILINE_STRING (Passed):** 
1. I installed the keymandeveloper-18.0.133.exe build in Windows 10.
2. Created a new "Lexical Model" project.
3. Navigate to the "Models" tab and then click the "*.model.ts" link button.
4. Copy a word from a notepad file.
5. Click in the "Add word..." box.
6. Paste the text by pressing the CTRL+V key.
7. Verified that the "Confirm" dialog appears with "Yes", "No" and Cancel.
8. Click the "Yes" button.
9. Verified that the word appears with multiple rows. No errors were displayed.
It works well.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.135-alpha](https://keyman.com/downloads/releases/alpha/18.0.135)

**Pull request merged**